src/engine/SpatialJoin.h

```diff
@@ -135,5 +141,5 @@ class SpatialJoin : public Operation {
   // between the two objects
   bool addDistToResult_ = true;
   const string nameDistanceInternal_ = "?distOfTheTwoObjectsAddedInternally";
-  bool useBaselineAlgorithm_ = false;
+  Algorithm algorithm_ = Algorithm::BoundingBox;
```

Comment:
Can you leave the default at S2?
The other student is working on a syntax that will make it easy to choose the algorithm.
We can then experiment with the performance characteristics of these algorithms.